core_backend/app/database.py

```diff
@@ -70,3 +70,10 @@ async def get_async_session() -> AsyncGenerator[AsyncSession, None]:
         get_sqlalchemy_async_engine(), expire_on_commit=False
     ) as async_session:
         yield async_session
+
+
+async def get_async_session_direct() -> AsyncSession:
+    """Return a SQLAlchemy async session."""
+    engine = get_sqlalchemy_async_engine()
+    async_session = AsyncSession(engine, expire_on_commit=False)
+    return async_session
```

Comment:
Any ideas of how to clean this up somehow? Why do we use generators anyway? Was having a lot of trouble with the generators so made this new function to use with auth. Please suggest better approaches!

Reply:
Using the context manager makes sure that the session is always closed when done. The level where it is created is the level where it should be closed - that is best practice. You should not depend on the function calling it to clean up properly.

But in order to do that, we need to return control to the context manager. And you need a generator to do that.

I'll take a look where it is being used and see if I can suggest other ways to do it.

Reply:
Thanks for the clarification!